include/boost/geometry/iterators/closing_iterator.hpp

```diff
@@ -38,9 +38,12 @@ struct closing_iterator
     <
         closing_iterator<Range>,
         typename boost::range_value<Range>::type const,
-        boost::random_access_traversal_tag
+        boost::random_access_traversal_tag,
+        typename boost::range_reference<Range>::type const,
+        typename boost::range_difference<Range>::type
     >
 {
+    typedef typename boost::range_reference<Range>::type const reference_type;
```

Comment:
I think the proper way to define those is to use the types from the `iterator_facade` class.
Same for `difference_type`.

Reply:
It really does not matter: I just checked the ``iterator_facade`` class and the types you provide are those used to define ``iterator_facade::reference`` and ``iterator_facade::difference_type``. So you code is perfectly fine.

My comment should be understood as a stylistic one.

Reply:
Yep, totally agree. Fixed.